### PR TITLE
Define `BitmovinCastManager` API

### PR DESCRIPTION
**Pull request by rolandkakonyi**
## Description

We need to expose `BitmovinCastManager` APIs to enable casting features.

## Changes

Introduced `BitmovinCastManager` and its APIs.

## Checklist
- [x] 🗒 `CHANGELOG` entry - already exists in #239

### CODE REVIEW

File: src/bitmovinCastManager.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+   *
+   * @param options The options to be used for initializing `BitmovinCastManager`
+   * @returns A promise that resolves when the `BitmovinCastManager` was initialized successfully
+   */
+  initialize: async (
+    options: BitmovinCastManagerOptions | null = null
+  ): Promise<void> => {
+    if (Platform.OS === 'ios' && Platform.isTV) {
+      return Promise.resolve();
+    }
```
Comment (strangesource) [inline src/bitmovinCastManager.ts:54]: Should we document this on the function or even the interface?

File: src/bitmovinCastManager.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+   * The provided metadata must be JSON serializable.
+   * @param metadata The metadata to be sent
+   * @returns A promise that resolves when the metadata was sent successfully
+   */
+  sendMetadata: (metadata: Record<string, any>) => {
+    if (Platform.OS === 'ios' && Platform.isTV) {
+      return Promise.resolve();
+    }
+    return BitmovinCastManagerModule.sendMetadata(metadata);
+  },
```
Comment (strangesource) [inline src/bitmovinCastManager.ts:83]: I'd rather not include this. The functionality is deprecated on Android as it requires serialization of user objects - which is difficult and sometimes not intuitive on Android.
While this is maybe more valid in the React Native context, the serialization can be easily done by the customer and they can use `sendMessage` then.

**Review by strangesource — COMMENTED** [review]
Nice, just one remark and some minor NIT.

File: src/bitmovinCastManager.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+   * A custom message namespace to be used for communication between sender and receiver.
+   * Using `nil` value will result in using the default message namespace
+   */
+  messageNamespace?: string | null;
+}
+
+/**
+ * Singleton providing access to GoogleCast related features.
+ * The `BitmovinCastManager` needs to be initialized by calling `BitmovinCastManager.initialize`
+ * before `Player` creation to enable casting features.
```
[the remark this answers is not in the document]
Comment (matamegger) [reply src/bitmovinCastManager.ts:25]: Seems fine 👍 .
 On Android there is an auto initialization, but explicitly initializing always seems like a good idea.

File: src/bitmovinCastManager.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+const BitmovinCastManagerModule = NativeModules.BitmovinCastManagerModule;
+
+/**
+ * The options to be used for initializing `BitmovinCastManager`
+ * @platform Android, iOS
+ */
+export interface BitmovinCastManagerOptions {
+  /**
+   * ID of receiver application.
+   * Using `nil` value will result in using the default application ID
```
[the remark this answers is not in the document]
Comment (rolandkakonyi) [reply src/bitmovinCastManager.ts]: 🤦‍♂️, fixed in 3ee1aef

File: src/bitmovinCastManager.ts [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+   */
+  applicationId?: string | null;
+  /**
+   * A custom message namespace to be used for communication between sender and receiver.
+   * Using `nil` value will result in using the default message namespace
```
[the remark this answers is not in the document]
Comment (rolandkakonyi) [reply src/bitmovinCastManager.ts]: same, fixed in 3ee1aef

File: src/bitmovinCastManager.ts (hunk 1 above)
Comment (rolandkakonyi) [reply src/bitmovinCastManager.ts:54]: @strangesource It is documented by `@platform Android, iOS` on the type.
This is for safety to not get into weird errors on tvOS.

Comment (strangesource) [reply src/bitmovinCastManager.ts:54]: Ah, missed that this is part of the API. 👍

File: src/bitmovinCastManager.ts (hunk 2 above)
Comment (rolandkakonyi) [reply src/bitmovinCastManager.ts]: OK, let's skip this one. Removed in: bcbaf4d

**Review by strangesource — APPROVED** [review]

**Pull request merged**